### ISSUE TITLE
Add 'scalar' `label_mode` in `image_dataset_from_directory` and load images from directory root for Regression

### ISSUE THREAD

**Issue by eduardofv** [keras-team/keras#15440]
**System information**.

TensorFlow version (you are using): 2.5,2.6
Are you willing to contribute it (Yes/No) : Yes

**Describe the feature and the current behavior/state**.

`image_dataset_from_directory` currently assumes that the dataset will be used in classification tasks, so the labels take some form of categorical values (integers, binary, etc). It also make strong assumptions on the directory structure in which the images are located: if a list of labels is passed, the images should be placed on subdirs named the same as the labels (even if the documentations says that the structure is ignored).

For regression purposes, I suggest a new `label_mode` called "scalar" in which labels can take a real value and are passed in `labels` as a list or tuple, and all images are located in the root of `directory`. 

Example: 
You have a directory with images and a text file with their labels for regression which are floats (labels are ordered alphanumerically according to image name):

```
with open("labels.txt") as fin:
  labels = [float(item) for item in fin.readlines()]
data = image_dataset_from_directory("images_dir", labels=labels, label_mode='scalar')
```

**Will this change the current api? How?**

The parameter `label_mode` will accept a new value `scalar`. This value will be passed to `index_directory` as a new parameter. Inside it, if `label_mode` is `scalar` then the directory indexing is not performed on the subdirs, only on root. 

**Who will benefit from this feature?**

Regression problems which uses images in a single directory

- Do you want to contribute a PR? (yes/no): 
  yes - I've already have a working version. If accepted, I'll do the Pull Request.

- Briefly describe your candidate solution(if contributing):

Beside tests, documentation and some validation these are the main changes:

image_dataset.py
```
def image_dataset_from_directory(
...
  image_paths, labels, class_names = dataset_utils.index_directory(
      directory,
      labels,
      formats=ALLOWLIST_FORMATS,
      class_names=class_names,
      shuffle=shuffle,
      seed=seed,
      follow_links=follow_links,
      label_mode=label_mode) #calling with new param
...
```

dataset_utils.py
```
def index_directory(directory,
                    labels,
                    formats,
                    class_names=None,
                    shuffle=True,
                    seed=None,
                    follow_links=False,
                    label_mode="int"): #mode parameter added
...
if labels is None or label_mode == 'scalar':
    # in the no-label case, index from the parent directory down.
    subdirs = ['']
    class_names = subdirs
  else:
...
```
